crates/ide_assists/src/handlers/extract_struct_from_enum_variant.rs

```diff
@@ -243,26 +314,46 @@ fn create_struct_def(
     strukt
 }
 
-fn update_variant(variant: &ast::Variant, generic: Option<ast::GenericParamList>) -> Option<()> {
+fn update_variant(variant: &ast::Variant, generics: Option<ast::GenericParamList>) -> Option<()> {
     let name = variant.name()?;
-    let ty = match generic {
-        // FIXME: This uses all the generic params of the enum, but the variant might not use all of them.
-        Some(gpl) => {
-            let gpl = gpl.clone_for_update();
-            gpl.generic_params().for_each(|gp| {
-                let tbl = match gp {
-                    ast::GenericParam::LifetimeParam(it) => it.type_bound_list(),
-                    ast::GenericParam::TypeParam(it) => it.type_bound_list(),
-                    ast::GenericParam::ConstParam(_) => return,
-                };
-                if let Some(tbl) = tbl {
-                    tbl.remove();
-                }
-            });
-            make::ty(&format!("{}<{}>", name.text(), gpl.generic_params().join(", ")))
-        }
-        None => make::ty(&name.text()),
-    };
+    let ty = generics
+        .filter(|generics| generics.generic_params().count() > 0)
+        .map(|generics| {
+            let generic_str = generics
+                .generic_params()
+                .with_position()
+                .map(|p| match p {
+                    itertools::Position::First(p) | itertools::Position::Middle(p) => (p, true),
+                    itertools::Position::Last(p) | itertools::Position::Only(p) => (p, false),
+                })
+                .fold(String::with_capacity(8), |mut s, (p, more)| {
```

Comment:
Instead of folding over the String here, it would probably read nicer if just looped over the iterator and push into the string that way(either via a for loop or for_each whatever you prefer)